### ISSUE TITLE
setting window position sets the window decoration pos instead of client area pos

### ISSUE THREAD

**Issue by andrewrk** [glfw/glfw#626]
Here's my test (against 43f4afef which is current master):

```diff
diff --git a/examples/simple.c b/examples/simple.c
index 37cf105..ff82ede 100644
--- a/examples/simple.c
+++ b/examples/simple.c
@@ -30,6 +30,9 @@
 #include <stdlib.h>
 #include <stdio.h>
 
+int hardcoded_left = 50;
+int hardcoded_top = 50;
+
 static void error_callback(int error, const char* description)
 {
     fputs(description, stderr);
@@ -41,6 +44,10 @@ static void key_callback(GLFWwindow* window, int key, int scancode, int action,
         glfwSetWindowShouldClose(window, GLFW_TRUE);
 }
 
+static void window_pos_callback(GLFWwindow* window, int left, int top) {
+    fprintf(stderr, "window_pos_callback %d %d\n", left, top);
+}
+
 int main(void)
 {
     GLFWwindow* window;
@@ -57,6 +64,15 @@ int main(void)
         exit(EXIT_FAILURE);
     }
 
+    fprintf(stderr, "set to %d %d\n", hardcoded_left, hardcoded_top);
+    glfwSetWindowPos(window, hardcoded_left, hardcoded_top);
+
+    int left, top;
+    glfwGetWindowPos(window, &left, &top);
+    fprintf(stderr, "get pos %d %d\n", left, top);
+
+    glfwSetWindowPosCallback(window, window_pos_callback);
+
     glfwMakeContextCurrent(window);
     gladLoadGLLoader((GLADloadproc) glfwGetProcAddress);
     glfwSwapInterval(1);

```

Output:

```
set to 50 50
get pos -55 -69
window_pos_callback 5 19
window_pos_callback 55 69
```

Based on the documentation, I expect `glfwSetWindowPos` to set the x, y location of the client area, not the window decoration area. So when I get the window pos, I expect to get 50, 50 back instead of -55, -69, and then when I get the window pos callback, I expect to again get 50, 50, instead of 55, 69.

I noticed that if I moved the `glfwSetWindowPos` call to the key callback, it worked as intended.

In my application, I'm storing the window position in a configuration file, and this bug causes the window to be offset by an additional 5, 19 every time the user opens the application.

System information:

```
Linux nixos 4.1.8 #1-NixOS SMP Thu Jan 1 00:00:01 UTC 1970 x86_64 GNU/Linux
XFCE 4.12
```

**Comment by andrewrk** [comment]
As a workaround, I can set a flag to determine when window pos callback is first called, and then call `glfwSetWindowPos` in that callback.

**Comment by elmindreda** [comment]
This is a duplicate of #517.

**Comment by elmindreda** [comment]
The bogus positions passed to the position callback should be fixed with 6494da3101eba2ad0d9c8b814e98465456c11273. As for the rest, can it still be reproduced? I was unable to reproduce #274 with Xfwm4 4.12 on Ubuntu 15.10 today.
